inc/import/parser/class.ParserFITMultiple.php

```diff
@@ -11,101 +11,78 @@
 /**
  * Abstract parser for multiple activities in *.fit-file
  *
- * @author Hannes Christiansen
+ * @author undertrained
  * @package Runalyze\Import\Parser
  */
 class ParserFITMultiple extends ParserAbstractMultiple {
-	/** @var string */
-	const PERL_FIT_ERROR_MESSAGE_START = 'main::Garmin::FIT';
-
-	/** @var string */
-	const PERL_GENERAL_MESSAGE_START = 'perl: warning:';
-
 	/**
-	 * Name of output file
-	 * @var string
+	 * Fit Data
+	 * @var adriangibbons\phpFITFileAnalysis
 	 */
-	protected $Filename = '';
+	protected $fitData = null;
 
 	/**
-	 * Handle
-	 * @var resource
+	 * Set fit data
+	 * @param object $fit
 	 */
-	protected $Handle = null;
-
-	/**
-	 * Set filename
-	 * @param string $filename
-	 */
-	public function setFilename($filename) {
-		$this->Filename = $filename;
+	public function setFitData($fit) {
+		$this->fitData = $fit;
 	}
 
 	/**
 	 * Parse
 	 */
 	public function parse() {
-		$firstActivityStarted = false;
-		$SingleParser = new ParserFITSingle('');
+		if (isset($this->fitData->data_mesgs['activity']['num_sessions']) &&
+		    $this->fitData->data_mesgs['activity']['num_sessions'] > 1) {
+			for ($session = 0; $session < $this->fitData->data_mesgs['activity']['num_sessions']; $session++) {
+				$fitsingle = clone $this->fitData;
+
+				if (isset($fitsingle->data_mesgs['session'])) {
+					if (isset($fitsingle->data_mesgs['session']['start_time'][$session]) &&
+					    isset($fitsingle->data_mesgs['session']['total_elapsed_time'][$session])) {
+						$start_time = $fitsingle->data_mesgs['session']['start_time'][$session];
+						$end_time = $fitsingle->data_mesgs['session']['total_elapsed_time'][$session] + $start_time;
+
+						if (isset($fitsingle->data_mesgs['record'])) {
+							foreach (array_keys($fitsingle->data_mesgs['record']) as $key) {
```

Comment:
As you are accessing the resulting data via `$fitsingle->data_mesgs['record'][$key]` afterwards, you can use
```php
foreach ($fitsingle->data_mesgs['record'] as $key => $record) {
    if ('timestamp' == $key) {
        foreach ($record as $recordKey => $recordValue) {
            if ($recordValue < $start_time || $recordValue > $end_time) {
                unset($fitsingle->data_mesgs['record'][$key][$recordKey]);
            }
        }
    } else {
        // similar ...
    }
}
```